### PR TITLE
[15.0][FIX] account_invoice_report_grouped_by_picking: sold service r…

### PR DESCRIPTION
**Pull request by stefan-tecnativa**
…efund

cc @Tecnativa TT44067

@chienandalu please review!

This PR is a FW Port of the changes done in 13: #278

### CODE REVIEW

**Comment by chienandalu** [comment]
You should Fw to v14 first, don't you think?

**Comment by chienandalu** [comment]
And why are there only on commit. Just cherry pick one by one and resolve conflicts if any

File: account_invoice_report_grouped_by_picking/models/account_move.py [hunk 1]
```diff
@@ -113,17 +113,19 @@ def lines_grouped_by_picking(self):
...
                 remaining_qty = 0.0
                 for key in picking_dict:
                     picking_dict[key] = abs(picking_dict[key])
             if not float_is_zero(
                 remaining_qty,
                 precision_rounding=line.product_id.uom_id.rounding or 0.01,
             ):
-                self._process_section_note_lines_grouped(
-                    previous_section, previous_note, lines_dict
-                )
```
Comment (carlosdauden) [inline account_invoice_report_grouped_by_picking/models/account_move.py:126]: Original commit don't remove this, but It is moved inside:
`elif not line.move_line_ids and not line.sale_line_ids:`
at line 110

Comment (carlosdauden) [inline account_invoice_report_grouped_by_picking/models/account_move.py:126]: Original commit don't remove this, but It is moved inside:
`elif not line.move_line_ids and not line.sale_line_ids:`
at line 110
¿Why is it removed?

**Review by carlosdauden — CHANGES_REQUESTED** [review]

File: account_invoice_report_grouped_by_picking/models/account_move.py (hunk 1 above)
Comment (chienandalu) [reply account_invoice_report_grouped_by_picking/models/account_move.py:126]: Maybe a merge conflict not so well resolved?

Comment (stefan-tecnativa) [reply account_invoice_report_grouped_by_picking/models/account_move.py:126]: Yes, I missed that during the merge. Thanks for the comment.

**Review by carlosdauden — APPROVED** [review]

File: account_invoice_report_grouped_by_picking/models/account_move.py [hunk 2]
```diff
@@ -117,7 +117,7 @@ def lines_grouped_by_picking(self):
             # To avoid to print duplicate lines because the invoice is a refund
             # without returned goods to refund.
             if (
-                self.type == "out_refund"
+                self.move_type == "out_refund"
```
Comment (chienandalu) [inline account_invoice_report_grouped_by_picking/models/account_move.py:120]: This fix should go in the orginal commit

File: account_invoice_report_grouped_by_picking/tests/test_account_invoice_group_picking.py [hunk 3]
```diff
@@ -327,10 +327,10 @@ def test_account_invoice_group_picking_service_sale_reports_check(self):
         self.sale2.action_confirm()
         # invoice sales
         invoice = self.sale2._create_invoices()
-        invoice.post()
+        invoice._post()
         # Test report
         content = html.document_fromstring(
-            self.env.ref("account.account_invoices").render_qweb_html(invoice.id)[0]
+            self.env.ref("account.account_invoices")._render_qweb_html(invoice.id)[0]
```
Comment (chienandalu) [inline account_invoice_report_grouped_by_picking/tests/test_account_invoice_group_picking.py:333]: And this changes should go to your tests commit

**Review by chienandalu — CHANGES_REQUESTED** [review]

**Review by chienandalu — APPROVED** [review]
Thanks, now it's correct :)

**Pull request merged**